packages/react-admin/src/mui/input/SelectArrayInput.js

```diff
@@ -0,0 +1,260 @@
+ * the related choice as the `record` prop. You can use Field components there.
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const FullNameField = ({ record }) => <span>{record.first_name} {record.last_name}</span>;
+ * <SelectArrayInput source="authors" choices={choices} optionText={<FullNameField />}/>
+ *
+ * The choices are translated by default, so you can use translation identifiers as choices:
+ * @example
+ * const choices = [
+ *    { id: 'programming', name: 'myroot.tags.programming' },
+ *    { id: 'lifestyle', name: 'myroot.tags.lifestyle' },
+ *    { id: 'photography', name: 'myroot.tags.photography' },
+ * ];
+ */
+export class SelectArrayInput extends Component {
+    /*
+     * Using state to bypass a redux-form comparison but which prevents re-rendering
+     * @see https://github.com/erikras/redux-form/issues/2456
+     */
+    state = {
+        value: this.props.input.value || [],
+    };
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.input.value !== this.props.input.value) {
+            this.setState({ value: nextProps.input.value || [] });
+        }
+    }
+
+    handleChange = event => {
+        this.props.input.onChange(event.target.value);
+        this.setState({ value: event.target.value });
+    };
+
+    renderMenuItem = choice => {
+        const { optionText, optionValue, translate } = this.props;
+        const choiceName = React.isValidElement(optionText) // eslint-disable-line no-nested-ternary
+            ? React.cloneElement(optionText, { record: choice })
+            : typeof optionText === 'function'
+              ? optionText(choice)
+              : get(choice, optionText);
+        return (
+            <MenuItem
+                key={get(choice, optionValue)}
+                value={get(choice, optionValue)}
+            >
+                {translate(choiceName, { _: choiceName })}
+            </MenuItem>
+        );
+    };
+
+    render() {
+        const {
+            choices,
+            classes,
+            className,
+            isRequired,
+            label,
+            meta,
+            options,
+            resource,
+            source,
+            optionText,
+            optionValue,
+            ...rest
+        } = this.props;
+        if (typeof meta === 'undefined') {
+            throw new Error(
+                "The SelectInput component wasn't called within a redux-form <Field>. Did you decorate it and forget to add the addField prop to your component? See https://marmelab.com/react-admin/Inputs.html#writing-your-own-input-component for details."
+            );
+        }
+        const { touched, error } = meta;
+
+        return (
+            <FormControl
+                className={classnames(classes.root, className)}
+                {...sanitizeRestProps(rest)}
```

Comment:
Am I right to think that is the user adds a prop to `SelectArrayInput` that she wants to pass to  to the `Select`, the `FormControl` will pass it automatically?

Reply:
No, as with any other components, you'll have to use the `options` props